docs/dev-guide/contrib-guides-taskfiles.md

```diff
@@ -278,5 +278,33 @@ my-task:
     - "lib2"
 ```
 
+## Task docstrings
+
+For reusable utility tasks that take arguments, you should add a docstring above the task definition
+using a JSDoc-like syntax. The following example demonstrates the syntax:
+
+```yaml
+# Task description including a description of any side effects or outputs generated by this task.
+#
+# @param {param_type} PARAM_1 Description for a required parameter.
+# @param {param_type} [PARAM_2] Description for an optional parameter.
+# @param {param_type} [PARAM_3=default_value] Description for an optional parameter with a default
+# value.
+# @param {[]param_type} PARAM_4 Description for a required parameter that's an array.
+# @param {map[key_type]value_type} PARAM_5 Description for a required parameter that's a map.
+```
+
+* The task's description (and the empty line following it) may be omitted if the task's name makes
+  it obvious.
+* Each parameter's description may be omitted if it's obvious.
```

Comment:
It might be clear enough as is, but maybe it is worth making it more explicit that the pragma, type, and name are mandatory even when the description is omitted?